src/responses.rs

```diff
@@ -299,15 +309,67 @@ impl EncryptionInfo {
         }
         .into()
     }
+}
+
+/// Struct containing information on how a to-device message was decrypted.
+#[wasm_bindgen()]
+#[derive(Debug, Clone)]
+pub struct OlmEncryptionInfo {
+    inner: matrix_sdk_common::deserialized_responses::EncryptionInfo,
+}
 
-    /// Get the algorithm used for encryption.
-    #[wasm_bindgen(getter, js_name = "algorithm")]
-    pub fn algorithm(&self) -> EncryptionAlgorithm {
-        match self.inner.algorithm_info {
-            AlgorithmInfo::MegolmV1AesSha2 { .. } => EncryptionAlgorithm::MegolmV1AesSha2,
-            AlgorithmInfo::OlmV1Curve25519AesSha2 { .. } => {
-                EncryptionAlgorithm::OlmV1Curve25519AesSha2
+impl TryFrom<matrix_sdk_common::deserialized_responses::EncryptionInfo> for OlmEncryptionInfo {
+    type Error = JsError;
+
+    fn try_from(
+        value: matrix_sdk_common::deserialized_responses::EncryptionInfo,
+    ) -> Result<Self, Self::Error> {
+        match value.algorithm_info {
+            AlgorithmInfo::MegolmV1AesSha2 { .. } => Err(JsError::new(
+                "AlgorithmInfo::MegolmV1AesSha2 is not applicable for OlmEncryptionInfo",
+            )),
+            AlgorithmInfo::OlmV1Curve25519AesSha2 { .. } => Ok(Self { inner: value }),
+        }
+    }
+}
+
+#[wasm_bindgen()]
+impl OlmEncryptionInfo {
+    /// The user ID of the to-device message sender. Note this is untrusted data
+    /// unless `isSenderVerified` is also true.
+    #[wasm_bindgen(getter)]
+    pub fn sender(&self) -> identifiers::UserId {
+        identifiers::UserId::from(self.inner.sender.clone())
+    }
+
+    /// The device ID of the device that sent us the to-device message.
+    /// Could be None in the case where the to-device message sender checks are
+    /// delayed (for room keys for example). For custom to-devices there is
+    /// no delay, this will not be None (the decryption would fail if the
+    /// sender device keys cannot be found).
+    /// Note this is untrusted data unless `isSenderVerified` is also true.
+    #[wasm_bindgen(getter, js_name = "senderDevice")]
+    pub fn sender_device(&self) -> Option<identifiers::DeviceId> {
+        Some(self.inner.sender_device.as_ref()?.clone().into())
+    }
+
+    /// The Curve25519 key of the device that encrypted the message.
+    #[wasm_bindgen(getter, js_name = "senderCurve25519Key")]
+    pub fn sender_curve25519_key(&self) -> Option<JsString> {
+        match &self.inner.algorithm_info {
+            AlgorithmInfo::MegolmV1AesSha2 { .. } => {
+                panic!("MegolmV1AesSha2 variant cannot be converted to OlmEncryptionInfo")
+            }
+            AlgorithmInfo::OlmV1Curve25519AesSha2 { curve25519_public_key_base64 } => {
+                Some(curve25519_public_key_base64.clone().into())
             }
         }
```

Comment:
it would probably be better to copy the `curve25519_public_key_base64` out of `algorithm_info` in `TryFrom<matrix_sdk_common::deserialized_responses::EncryptionInfo>`, to avoid this test and `panic`.

And since you're doing that, maybe consider copying `sender`, `verification_state`, etc.